brave-sns/src/main/java/smartthings/brave/sns/PublishRequestTracingHandler.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2016-2017 SmartThings
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package smartthings.brave.sns;
+
+import brave.Span;
+import brave.Tracing;
+import brave.propagation.TraceContext;
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.handlers.RequestHandler2;
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+
+import zipkin2.Endpoint;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * AWS RequestHandler2 that adds tracing around the afterError and afterResponse events
+ * when calling the AWS SNS publish API.
+ */
+public class PublishRequestTracingHandler extends RequestHandler2 implements SNSRequestTracingHandler {
```

Comment:
I haven't looked at the RequestHandler2 class much nor the X-Ray integrations with the AWS SDK.  From what I gather though we might want this to be a place where we hook in more Zipkin tracing details right?  If thats the case maybe this should be named `TracingRequestHandler` with the assumption more than just `PublishRequest` handling will be added here as well.

Reply:
Should be `final`